smac/intensification/intensification.py

```diff
@@ -307,7 +163,242 @@ def intensify(self, challengers, incumbent, run_history, aggregate_func,
+            incumbent : Configuration
+                best configuration so far
+            run_history : RunHistory
+                stores all runs we ran so far
+            aggregate_func: typing.Callable
+                aggregate performance across instances
+
+            Returns
+            -------
+            new_incumbent: Configuration
+                either challenger or incumbent
+        '''
+        # at least one run of challenger
+        # to increase chall_indx counter
+        first_run = False
+
+        # Line 8
+        N = max(1, self.minR)
+        
+        inc_inst_seeds = set(run_history.get_runs_for_config(incumbent))
+        # Line 9
+        while True:
+            chall_inst_seeds = set(run_history.get_runs_for_config(challenger))
+            
+            # Line 10
+            missing_runs = list(inc_inst_seeds - chall_inst_seeds)
+
+            # Line 11
+            self.rs.shuffle(missing_runs)
+            to_run = missing_runs[:min(N, len(missing_runs))]
+            # Line 13 (Line 12 comes below...)
+            missing_runs = missing_runs[min(N, len(missing_runs)):]
+
+            inst_seed_pairs = list(inc_inst_seeds - set(missing_runs))
+
+            inc_sum_cost = sum_cost(config=incumbent, instance_seed_pairs=inst_seed_pairs,
+                                    run_history=run_history)
+
+            # Line 12
+            for instance, seed in to_run:
+                # Run challenger on all <config,seed> to run
+                if self.run_obj_time:
+                    chall_inst_seeds = set(map(lambda x: (
+                        x.instance, x.seed), run_history.get_runs_for_config(challenger)))
+                    chal_sum_cost = sum_cost(config=challenger, instance_seed_pairs=chall_inst_seeds,
+                                             run_history=run_history)
+                    cutoff = min(self.cutoff,
+                                 inc_sum_cost *
+                                 self.Adaptive_Capping_Slackfactor
+                                 - chal_sum_cost
+                                 )
+
+                    if cutoff < 0:  # no time left to validate challenger
+                        self.logger.debug(
+                            "Stop challenger itensification due to adaptive capping.")
+                        break
+
+                else:
+                    cutoff = self.cutoff
+
+                if not first_run:
+                    first_run = True
+                    self._chall_indx += 1
+
+                self.logger.debug("Add run of challenger")
+                status, cost, dur, res = self.tae_runner.start(
+                    config=challenger,
+                    instance=instance,
+                    seed=seed,
+                    cutoff=cutoff,
+                    instance_specific=self.instance_specifics.get(instance, "0"))
+                self._num_run += 1
+
+                if status == StatusType.ABORT:
```

Comment:
Yes, it is new and makes the code later easier.
Right now, the ABORT only stops the intensification of the current challenger. 

How, do we want to exit if SMAC saw an ABORT?
Directly, sys.exit(42)?